gui/src/Layout.js

```diff
@@ -156,8 +156,9 @@ class Layout extends Component {
               <Switch>
                 <Route exact path='/job_specs/:jobSpecId' component={JobSpec} />
                 <Route exact path='/job_specs/:jobSpecId/runs' component={JobSpecRuns} />
-                <Route exact path='/job_specs/:jobSpecId/runs/:jobRunId' component={JobSpecRun} />
-                <Route exact path='/config' component={Configuration} />
+                <Route exact path='/job_specs/:jobSpecId/runs/page/:jobRunsPage' component={JobSpecRuns} />
+                <Route exact path='/job_specs/:jobSpecId/runs/id/:jobRunId' component={JobSpecRun} />
```

Comment:
I understand that react conventions place parameters like `:jobRunsPage` inside the url as you've shown rather than as query parameters:
```
/job_specs/:jobSpecId/runs/page/:jobRunsPage
/job_specs/:jobSpecId/runs/id/:jobRunId
```

I was originally concerned with the show page using the arbitrary `/id/:jobRunId` to differentiate it from the paginated index pages, but it's growing on me. Here are some other ideas

```js
/job_specs/:jobSpecId/runs/pages/:jobRunsPage
/runs/:jobRunId // flatten the url in the show, might not be worth the refactor
```

```js
/job_specs/:jobSpecId/runs/pages/:jobRunsPage
/job_specs/:jobSpecId/runs/:jobRunId
```

These urls will set the precedent for future paginated results so lets get them right the first time 😉

Reply:
I like
```
/job_specs/:jobSpecId/runs/pages/:jobRunsPage
/job_specs/:jobSpecId/runs/:jobRunId
```
Might need to hear what @rupurt has to say about this.